services/efs-guardian/src/simcore_service_efs_guardian/services/efs_manager.py

```diff
@@ -90,3 +94,44 @@ async def remove_project_node_data_write_permissions(
         )
 
         await efs_manager_utils.remove_write_permissions_bash_async(_dir_path)
+
+    async def list_projects_across_whole_efs(self) -> list[ProjectID]:
+        _dir_path = self._efs_mounted_path / self._project_specific_data_base_directory
+
+        # List all items in the directory
+        items = os.listdir(_dir_path)
+
+        # Filter and list only directories (which should be Project UUIDs)
+        directories = []
+        for item in items:
+            _item_path = _dir_path / item
+            if Path.is_dir(_item_path):
+                try:
+                    _project_id = parse_obj_as(ProjectID, item)
+                    directories.append(_project_id)
+                except ValueError:
+                    _logger.warning(
+                        "This is not a project ID. This should not happen! %s",
+                        _item_path,
+                    )
+            else:
+                _logger.warning(
+                    "This is not a directory. This should not happen! %s",
+                    _item_path,
+                )
+
+        return directories
+
+    async def remove_project_efs_data(self, project_id: ProjectID) -> None:
+        _dir_path = (
+            self._efs_mounted_path
+            / self._project_specific_data_base_directory
+            / f"{project_id}"
+        )
+
+        if Path.exists(_dir_path):
```

Comment:
there is no need to use class methods. your object is already a `Path`.
```python
if dir_path.exists():
    shutil.rmtree(dir_path)

```

- what if it is not a directory? how do you handle that?
- what if [rmtree fails](https://docs.python.org/3/library/shutil.html#shutil.rmtree) ... How do you handle e.g. there is no access rights to write in the disk?


I am not implying that you have to handle all errors here but rather check what is the impact in your service. e.g. is it compromising your background task? is the service crashing? etc

Reply:
improved :+1:

Reply:
I do not see any changes. Did you commit or i am looking at the wrong branch?

Reply:
I added some error handling there: 

```
            try:
                shutil.rmtree(_dir_path)
                _logger.info("%s has been deleted.", _dir_path)
            except FileNotFoundError as e:
                _logger.error("Directory %s does not exist.", _dir_path)
            except PermissionError as e:
                _logger.error("Permission denied when trying to delete %s.", _dir_path)
            except NotADirectoryError as e:
                _logger.error("%s is not a directory.", _dir_path)
            except OSError as e:
                _logger.error("Error: %s. Issue with path: %s", e, _dir_path)
```